### DISCUSSION TITLE
DB Browser for SQLite v3.13.1

### DISCUSSION THREAD

**Discussion by justinclift** [sqlitebrowser/sqlitebrowser#3775] [General]
This is a new release with several improvements over the v3.13.0 release from three months ago. :rocket:

As a special mention, SQLean has a new `time` extension! Check out the following link for more information: https://github.com/nalgeon/sqlean/blob/main/docs/time.md

Thanks to everyone for being part of our Community!

#### Added
- Add links to the wiki as help for many dialogs (7af1256, b5abc86)

#### Changed
- Change the default to install a shortcut to the Start menu when installing on Windows (623bec6)
- Improve icon image for links in 'Help' menu (#3693, 027c6a8)
- Pragma names and values can now be translated (#3697, bf62f3a, ad00ad4)
- Remove trailing characters when copying a single cell (#3735, 20f481a)
- Update AppImage binary to use the latest SQLCipher library (#3744, 21ba2d0)
- When editing the DB cells, expanded queries with parameters are included in the SQL Log (ac3209f)

#### Fixed
-  ExtendedTableWidget
    - Fix an issue that prevented autocomplete from working in some race conditions (#2567, #3706)
    - Use Tab to close autocomplete popup and move to next item (3aff8c9)
    - Troubleshoot 'Copy as SQL' action on a cell results in string being copied (#1952, 1ebe7bf)

- Global
    - Fix "Argument Missing" error in Korean translation (#3635, #3692, cd518de)
    - Fix an issue that caused apps to crash on some older versions of macOS (#3691)
    - Fix an issue when freezing after stopping a pragma (#3742, 15a9620)
    - Fix an issue with outputting the wrong version on macOS
    - Troubleshooting poor HiDPI support on Windows (#3684)
    - Troubleshoot SQL queries containing VACUUM that run incorrectly (#3723, 51784aa)

- MainWindow
    - Fix an issue when selecting queries containing multi-byte strings (#3731, f89097c)

- TableBrowser
    - Fix a bug where the first row was not adjusted (#3767, ad690e7)

- etc
    - Fix misleading links to Qt license information (8361aa5)
    - Replace 'http' with 'https' in the 'AboutDialog' and 'MainWindow' (9832a52)
    - Translation
        - German (#3749, 8e38bf0)
        - Indonesian (#3756, 6100595)
        - Japanese (#3755, e0f6aea)
        - Korean (7bbfcf9)
        - Simplified Chinese (#3761, 9eb0a5a)
        - Spanish (ac33918)

### Dependent library version information for each OS
|     **-**    |                     [**Qt**](https://www.qt.io/)                     |            [**SQLCipher**](https://www.zetetic.net/sqlcipher/)            |            [**SQLite**](https://sqlite.org/)            |         [**SQLean**](https://github.com/nalgeon/sqlean)         |
|:------------:|:--------------------------------------------------------------------:|:-------------------------------------------------------------------------:|:-------------------------------------------------------:|:---------------------------------------------------------------:|
| **AppImage** | [5.15.13](https://www.qt.io/blog/commercial-lts-qt-5.15.13-released) | [4.5.6](https://www.zetetic.net/blog/2024/01/17/sqlcipher-4.5.6-release/) | [3.46.1](https://www.sqlite.org/releaselog/3_46_1.html) |                          Not applicable                         |
|   **macOS**  | [5.15.13](https://www.qt.io/blog/commercial-lts-qt-5.15.13-released) | [4.6.1](https://www.zetetic.net/blog/2024/08/20/sqlcipher-4.6.1-release/) |                      Not applicable                     | [0.27.1](https://github.com/nalgeon/sqlean/releases/tag/0.27.1) |
|  **Windows** |          [5.15.2](https://www.qt.io/blog/qt-5.15.2-released)         | [4.6.1](https://www.zetetic.net/blog/2024/08/20/sqlcipher-4.6.1-release/) | [3.46.1](https://www.sqlite.org/releaselog/3_46_1.html) | [0.27.1](https://github.com/nalgeon/sqlean/releases/tag/0.27.1) |

### SHA256SUMS
- DB.Browser.for.SQLite-v3.13.1-win32.msi
    - e0b9f86d3da4d8d800e144295487e43de306c1bd27f14dccfe41e904736f25f7
- DB.Browser.for.SQLite-v3.13.1-win32.zip
    - 917ad2fa8d36e3bfa3fc85b11a34a8c18d189fbc2289f5a0d3bf41de8a288edc
- DB.Browser.for.SQLite-v3.13.1-win64.msi
    - d023d54b3a5db10c7e896089bb3dbe6e7f4bc4eaa9bbecb34ca414be5970f688
- DB.Browser.for.SQLite-v3.13.1-win64.zip
    - 22375e275ec42d96de1d3b8e9ea4ed86d2a3505c4d0ffcbd1af67aa4003e5e4d
- DB.Browser.for.SQLite-v3.13.1-x86.64.AppImage
    - d6563c5c211a73192da96e3bb11a3bf83a2f3164aa4db83482c0aecf8b751b77
- DB.Browser.for.SQLite-v3.13.1.dmg
    - a641cfbfcc2ce609f07de44a35134dab53485ecc18e6d9afa297b514d74bd75e

<hr /><em>This discussion was created from the release <a href='https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.13.1'>DB Browser for SQLite v3.13.1</a>.</em>

**Comment by moppman** [comment]
Just a heads-up: this release breaks AppImages on `Debian Stable` because https://github.com/sqlitebrowser/sqlitebrowser/commit/4d530dce9db74acf57068b8179cecebbf85a972f drew in a `libfuse2t64` dependency which is not avaiblable on `Debian Stable`.

I don't think this necessarily warrants a proper bug report because Debian users can just use the version from the official Debian repo.
Just thought I'd share here instead.

**Reply by justinclift** [reply]
Thanks for letting us know @moppman.  That commit jumped the GitHub Actions builder from Ubuntu LTS 20.04 to 24.04.

We could probably check if Ubuntu LTS 22.04 (the one in between) would get things working again or not.  If it does, we could make a 2nd version of the AppImage with that fix. :smile:

**Reply by FriedrichFroebel** [reply]
Adding on this, switching to Ubuntu 24.04 as the AppImage base has broken support for Ubuntu 22.04 as well:

```
me@localhost:~/Downloads$ ./DB.Browser.for.SQLite-v3.13.1-x86.64.AppImage 
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /tmp/.mount_DB.BrozwLGOh/AppRun.wrapped)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/AppRun.wrapped)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/AppRun.wrapped)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libsqlcipher.so.1)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libsqlcipher.so.1)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libQt5Network.so.5)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libQt5Gui.so.5)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libcrypto.so.3)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libgssapi_krb5.so.2)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libicui18n.so.74)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libicui18n.so.74)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libicuuc.so.74)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libicuuc.so.74)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libglib-2.0.so.0)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libkrb5.so.3)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libk5crypto.so.3)
/tmp/.mount_DB.BrozwLGOh/AppRun.wrapped: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_DB.BrozwLGOh/usr/bin/../lib/libkrb5support.so.0)
```

Ideally we would generate compatible builds until regular Ubuntu support ends, which is April 2025 for Ubuntu 20.04 for example: https://endoflife.date/ubuntu

**Reply by lucydodo** [reply]
Hi, I'm the author of that commit. I apologize for the inconvenience, this side-effect is something I hadn't thought of.
I'll try to fix it by the end of the week and deploy the new AppImage: #3781

**Reply by lucydodo** [reply]
Thank you for reporting this. We have released a new v2 version with this issue fixed: https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.13.1
> Confirmed working on Ubuntu 20.04.6.

**Comment by andrerav** [comment]
Hi,

FYI, some of the download links on the [release blog post](https://sqlitebrowser.org/blog/version-3-13-1-released/) appear to be broken. I tested the "Standard (MSI) installer for Win64" and ".zip (no installer) for Win64", both of which returns a 404. 

I found working download links here: https://download.sqlitebrowser.org/

**Reply by lucydodo** [reply]
Thank you for letting us know.
I missed it during the migration work a few months ago. The issue is now resolved.
